File: 1895-minimum-number-of-operations-to-move-all-balls-to-each-box/minimum-number-of-operations-to-move-all-balls-to-each-box.cpp
class Solution {
public:
    vector<int> minOperations(string boxes) {
        vector<int>res;
        vector<int>ind;
        for(int i=0;i<boxes.size();i++){
            if(boxes[i]=='1') ind.push_back(i);
        }
        for(int i=0;i<boxes.size();i++){
            int steps = 0;
            for(auto it:ind){
                steps += abs(i-it);
            }
            res.push_back(steps);
        }
        return res;
    }
};

File: 1895-minimum-number-of-operations-to-move-all-balls-to-each-box/minimum-number-of-operations-to-move-all-balls-to-each-box.rs
impl Solution {
    pub fn min_operations(boxes: String) -> Vec<i32> {
        let mut res = Vec::new();
        let mut indices = Vec::new();

        // Collect indices where boxes contain '1'
        for (i, ch) in boxes.chars().enumerate() {
            if ch == '1' {
                indices.push(i as i32);
            }
        }

        // Compute the number of steps for each box
        for i in 0..boxes.len() {
            let steps: i32 = indices.iter().map(|&ind| (i as i32 - ind).abs()).sum();
            res.push(steps);
        }

        res
    }
}
